### PR TITLE
Use epicsStrtod, remove some warnings

### PR DESCRIPTION
**Pull request by FreddieAkeroyd**
Use `epicsStrtod`

### CODE REVIEW

File: src/libCom/yajl/yajl_parser.c [hunk 1]
```diff
@@ -45,6 +45,8 @@
 #include "yajl_encode.h"
 #include "yajl_bytestack.h"
 
+#include <osdStrtod.h>
+
```
Comment (anjohnson) [inline src/libCom/yajl/yajl_parser.c:49]: I'd prefer that you replace the `#include <stdlib.h>` with `#include <epicsStdlib.h>` than pull in this header, which was really meant to be an implementation detail.

**Review by anjohnson — APPROVED** [review]
Needs care when merging up to 7.0 to avoid regressions since both `memneeded` and `YA_MALLOC()` have already been modified to use `size_t` there.

File: src/libCom/yajl/yajl_parser.c (hunk 1 above)
Comment (FreddieAkeroyd) [reply src/libCom/yajl/yajl_parser.c:49]: Thanks - sorry, I should have looked more closely how it was used elsewhere in the code

**Comment by FreddieAkeroyd** [comment]
@anjohnson Should I bring the YA_MALLOC (and any other 7.0 differences) back onto the 3.15 branch?

**Comment by anjohnson** [comment]
@FreddieAkeroyd No, that was part of a major reimport of the yajl code from upstream. This looks good now, thanks!

**Pull request merged**